org.eclipse.winery.edmm/src/main/java/org/eclipse/winery/edmm/model/EdmmImporter.java

```diff
@@ -0,0 +1,458 @@
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+public class EdmmImporter {
+
+    private final static Logger logger = LoggerFactory.getLogger(EdmmImporter.class);
+
+    private final Map<QName, TNodeType> nodeTypes;
+    private final Map<QName, TNodeTypeImplementation> nodeTypeImplementations;
+    private final Map<String, Map.Entry<QName, TNodeType>> normalizedNodeTypes = new HashMap<>();
+
+    private final Map<QName, TRelationshipType> relationshipTypes;
+
+    private final Map<String, Map.Entry<QName, TRelationshipType>> normalizedRelationshipTypes = new HashMap<>();
+    private final Map<QName, TRelationshipTypeImplementation> relationshipTypeImplementations;
+
+    private final Map<QName, TArtifactTemplate> artifactTemplates;
+    private final Map<String, Map.Entry<QName, TArtifactTemplate>> normalizedArtifactTemplates = new HashMap<>();
+
+    private final Map<EdmmType, QName> edmmToToscaMap;
+
+    private final IRepository repository;
+
+    private final String nodeTypesString = "nodeTypes";
+    private final String relationshipTypesString = "relationshipTypes";
+
+    public EdmmImporter() {
+        logger.debug("Initializing EDMM Importer...");
+
+        this.repository = RepositoryFactory.getRepository();
+
+        this.nodeTypes = repository.getQNameToElementMapping(NodeTypeId.class);
+        this.relationshipTypes = repository.getQNameToElementMapping(RelationshipTypeId.class);
+        this.nodeTypeImplementations = repository.getQNameToElementMapping(NodeTypeImplementationId.class);
+        this.relationshipTypeImplementations = repository.getQNameToElementMapping(RelationshipTypeImplementationId.class);
+        this.artifactTemplates = repository.getQNameToElementMapping(ArtifactTemplateId.class);
+
+        normalizeToscaTypes();
+
+        EdmmManager edmmManager = EdmmManager.forRepository(repository);
+        this.edmmToToscaMap = edmmManager.getEdmmOneToToscaMap();
+
+        logger.info("Initialized EDMM Importer!");
+    }
+
+    public void importFromStream(InputStream uploadedInputStream, String fileName, boolean overwrite) {
+        Path tempFile = null;
+        try {
+            tempFile = FileUtils.getTempDirectory().toPath().resolve(fileName);
+            Files.copy(uploadedInputStream, tempFile, REPLACE_EXISTING);
+
+            transform(tempFile, overwrite);
+        } catch (IOException e) {
+            logger.error("Cloud not save uploaded file!");
+            throw new RuntimeException(e);
+        } finally {
+            if (tempFile != null) {
+                try {
+                    FileUtils.forceDelete(tempFile.toFile());
+                } catch (IOException e) {
+                    logger.warn("Could not delete temp file!");
+                }
+            }
+        }
+    }
+
+    public boolean transform(Path edmmFilePath, boolean override) {
```

Comment:
just make sure you don't mistakenly  write "override" instead of "overwrite" in this class